pylint/extensions/docparams.py

```diff
@@ -636,26 +645,23 @@ def _handle_no_raise_doc(self, excs: set[str], node: nodes.FunctionDef) -> None:
         self._add_raise_message(excs, node)
 
     def _add_raise_message(
-        self, missing_excs: set[str], node: nodes.FunctionDef
+        self, missing_exceptions: set[str], node: nodes.FunctionDef
     ) -> None:
         """Adds a message on :param:`node` for the missing exception type.
 
-        :param missing_excs: A list of missing exception types.
-
+        :param missing_exceptions: A list of missing exception types.
         :param node: The node show the message on.
         """
         if node.is_abstract():
-            try:
-                missing_excs.remove("NotImplementedError")
-            except KeyError:
-                pass
-
-        if not missing_excs:
-            return
-
-        self.add_message(
-            "missing-raises-doc", args=(", ".join(sorted(missing_excs)),), node=node
-        )
+            with contextlib.suppress(KeyError):
```

Comment:
I would advise against this as this is much worse for performance. `try...except` have minimal performance impact, especially in `3.11`. With context manager you need look up the manager in the module, set it up, call the `__enter__`, call the `__exit__`, etc. 
There was recent discussion about this in the `CPython` repository about the additional costs.

Reply:
I reverted it, let's wait for the discussion on #7201to conclude before changing anything in pylint.

Reply:
@DanielNoord Could you link to the discussion please?

Reply:
https://github.com/python/cpython/issues/94933#issuecomment-1186581004

"Discussion" is perhaps too much 😅